### PR TITLE
Clean up some utility code

### PR DESCRIPTION
**Pull request by Hummer12007**

### CODE REVIEW

File: src/basic/path-util.h [hunk 1]
```diff
@@ -9,99 +9,20 @@
...
-int safe_getcwd(char **ret);
-int path_make_absolute_cwd(const char *p, char **ret);
 char* path_startswith(const char *path, const char *prefix) _pure_;
 int path_compare(const char *a, const char *b) _pure_;
 bool path_equal(const char *a, const char *b) _pure_;
 bool path_equal_or_files_same(const char *a, const char *b, int flags);
-char* path_join(const char *root, const char *path, const char *rest);
-char* path_simplify(char *path, bool kill_dots);
 
 /* Note: the search terminates on the first NULL item. */
```
Comment (kennylevinsen) [inline src/basic/path-util.h:19]: This comment belongs to PATH_IN_SET, so it should be removed as well.

Comment (Hummer12007) [reply src/basic/path-util.h]: Done

**Comment by kennylevinsen** [comment]
It would be nice to cleanup the history a little bit. Should be fine otherwise.

**Review by emersion — APPROVED** [review]
Thanks!

**Pull request merged**